### DISCUSSION TITLE
New WinUI3 Project (Packaged or Unpackaged) and Installed Windows Service?

### DISCUSSION THREAD

**Discussion by pjmagee** [microsoft/WindowsAppSDK#1047] [Q&A]
With so much documentation on MSIX, UwP and .NET5 around Background services or Tasks I'm struggling to know how to approach my problem.

Here is what I plan:

- A WinUI3 App that will:
  - Show a data grid of processed files from AppData produced by another Process (A game)
     - When files are detected, they will appear here with a status (pending/invalid/uploaded)
     - Detected files (FileSystemWatchers) are used to detect these files and are processed and uploaded for analysis.
     - Some other detected files are used for near real-time analysis for other features (sending http api calls to update other services to know about the current state)
  - Store certain processed files into a json file to keep history of previously processed files that was success/fail so when the app launches up, the user will see past files from previous sessions, or if the app was not being used for a few days, they will see the app begin to process files and upload them (If a Windows service was installed, maybe its always running?)
  - A settings page with various toggles to disable/enable certain features:
     - Enable/Disable features (which will change the behaviour of the Windows Service)
 - A session page that will show the realtime detected session of a running game based on monitored files
      - It will show some details of the current session based on the files it has detected and processed in the background

Can anyone point me in the right direction for something that can be a pure .NET5 solution with ProjectReunion 0.8?

I would like to continue down the MSIX route because packaging means it can be deployed to the App Store and that means releases are much easier, im just struggling to know how to side-install a Service that the UI can send/request data from a background Service App and allow the User to change or update features/settings that changes what the background Service should be doing.

**Comment by pjmagee** [comment]
Well, i think the best route i can take is going via Maui and using the Maui Startup and use the Hosting extensions for background work. Then I dont need to worry about seperate services running and communication between them.

**Reply by shelllet** [reply]
many many `win10` runs in environment where there is no network, i would like to continue down the none msix route.

**Reply by pjmagee** [reply]
> many many `win10` runs in environment where there is no network, i would like to continue down the none msix route.

MSIX requires an internet connection / network?

**Reply by shelllet** [reply]
`msix` package can only be published locally in test mode or published to the appstore. so i  say appstore requires an  internet connection to install msix package.

**Reply by shelllet** [reply]
i found [MSIX Packaging Tool](https://www.microsoft.com/en-us/p/msix-packaging-tool/9n5lw3jbcxkf?activetab=pivot:overviewtab) can to this.

**Reply by jonwis** [reply]
You can also use `.appinstaller` files to reference MSIXes on local storage (including "double click MSIX in explorer to install" behavior) and you can use `Add-AppXPackage` in PowerShell to deploy packages as well.  All of those are wrappers around the [PackageManager.AddPackageAsync](https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager.addpackageasync?view=winrt-20348) methods, which you can also call directly.

MSIX is just a packaging & deployment description format.  The Windows Store (including WinGet) are a way to _get_ MSIXes to deploy.

**Reply by jonwis** [reply]
(@jvintzel  might want to tag somebody else into this discussion, too)

**Reply by maxkatz6** [reply]
Unfortunately appinstaller is broken for a second year already, requiring users to reboot their PC to get new update, which makes it unusable. 
And I want to mention, that using these APIs directly from the your app works and could be the best solution for somebody.
